### PR TITLE
Fixes cases where new issues can duplicate orders, making them dissapear

### LINKED ISSUE #341: Card disappearing on reorder
Ran into a case where a care disappears on every move, then reappears  when a render is forced on on the milestones view:

uhoh

The card below meta-data (test-issue#7)
image

The card above meta-data (test-issue#6)
image

The meta-data of the card being moved:
image

### PR DESCRIPTION
**Pull request by discorick**
closes #341

### CODE REVIEW

File: ember-app/app/mixins/cards/card-move.js [hunk 1]
```diff
@@ -20,17 +20,19 @@ var CardMoveMixin = Ember.Mixin.create({
       var below_order = issue_below._data[this.get("orderKey")];
       return (above_order + below_order) / 2;
     },
-    moveToTop: function(issue_below){
-      var below_order = issue_below._data[this.get("orderKey")];
+    moveToTop: function(issue_below, key){
+      key = key ? key : this.get("orderKey");
```
Comment (dahlbyk) [inline ember-app/app/mixins/cards/card-move.js]: `key || this.get("orderKey")`

That said, can we just kill the `orderKey` property altogether? This implict statefulness seems a likely source of other bugs.

Comment (discorick) [inline ember-app/app/mixins/cards/card-move.js]: I am fairly certain we can. 👍

**Comment by dahlbyk** [comment]
To clarify my understanding, what's the expected behavior on Milestones when reordering from Tasks, and vice versa?

File: ember-app/app/mixins/cards/card-move.js (hunk 1 above)
Comment (discorick) [inline ember-app/app/mixins/cards/card-move.js]: Actually.. not so straight-forward - I just remembered why its there in the first place . It's not impossible to acheive but will probably cause more changes than I would hope for in this PR.

**Comment by discorick** [comment]
>  To clarify my understanding, what's the expected behavior on Milestones when reordering from Tasks, and vice versa?

Are you wanting to know the expected result from the creation of a new issue in both respective views? Or what is happening every time a card is moved.

**Comment by dahlbyk** [comment]
> Are you wanting to know the expected result from the creation of a new issue in both respective views? Or what is happening every time a card is moved.

Whatever is relevant to this PR? :grinning: 

To clarify, this regression was introduced in https://github.com/huboard/huboard-web/pull/314 when `cardMover.moveToTop()` was introduced with new usage patterns. The fix seems sound.

**Comment by discorick** [comment]
So, on the task view, when an issue is created the following happens:
  - The order of the top issue in the column is taken (if there is one) and the new issue's `order` key is set
  - Then, the column is resorted based on `milestone_order`, and the new issues `milestone_order` key is set based on the top issue
  - Finally we post and the issue is created. 

This is the same for issues created in the milestone view. The reason this bug was easy to miss is the module used to calculate the order of the issue would automatically choose the key (`milestone_order, or `order`) based on the current view, causing one of the calculations to be incorrect based on the view.

**Comment by dahlbyk** [comment]
Actual milestone order change looks good, but I'm sometimes seeing two issues rendering:
double-issue

**Comment by dahlbyk** [comment]
Behavior confirmed :+1:

**Pull request merged**